repos/system_upgrade/common/actors/createisorepofile/libraries/create_iso_repofile.py

```diff
@@ -0,0 +1,37 @@
+import os
+
+from leapp.libraries.stdlib import api
+from leapp.models import (
+    CustomTargetRepositoryFile,
+    TargetOSInstallationImage,
+)
+
+
+def produce_repofile_if_iso_used():
+    target_iso_msgs_iter = api.consume(TargetOSInstallationImage)
+    target_iso = next(target_iso_msgs_iter, None)
+
+    if not target_iso:
+        return
+
+    if next(target_iso_msgs_iter, None):
+        api.current_logger().warn('Received multiple TargetISInstallationImage messages, using the first one')
+
+    # Mounting was successful, create a repofile to copy into target userspace
+    repofile_entry_templace = ('[{repoid}]\n'
+                               'name={reponame}\n'
+                               'baseurl={baseurl}\n'
+                               'enabled=0\n'
+                               'gpgcheck=0\n')
+
+    repofile_content = ''
+    for repo in target_iso.repositories:
+        repofile_content += repofile_entry_templace.format(repoid=repo.repoid,
+                                                           reponame=repo.repoid,
+                                                           baseurl=repo.baseurl)
+
+    iso_repofile_path = os.path.join('/var/lib/leapp/', 'upgrade_iso.repo')
```

Comment:
Let's also add target OS version to the filename.

Reply:
I've changed the repo filename to `el8_iso.repofile` (or `el9` when upgrading to RHEL9), as the `upgrade` prefix was  needless.

Reply:
thanks